packages/next/next-server/server/load-components.ts

```diff
@@ -22,6 +24,11 @@ export type LoadComponentsReturnType = {
     revalidate?: number | boolean
   }
   unstable_getStaticPaths?: () => void
+  unstable_getServerProps?: (context: {
+    req: IncomingMessage
+    res: ServerResponse
+    query: ParsedUrlQuery
```

Comment:
What about `params`? IIRC we agreed we'd start to explicitly separate params from query in the new lifecycles.

Reply:
Updated